src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        this.size = 0;
+        this.head = null;
+        this.tail = null;
+    }
+
     @Override
     public void add(T value) {
+        addTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        int linkedListSize = size;
+        checkOnIndexOutOfBoundsException(index, false);
+
+        if (linkedListSize == 0) {
+            addHead(value);
+        } else if (linkedListSize == index) {
+            addTail(value);
+        } else {
+            Node<T> findedNode = findNode(index);
+            addBetween(value, findedNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            addTail(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkOnIndexOutOfBoundsException(index, true);
+        Node<T> needNode = findNode(index);
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/224542328-2a408206-e296-420c-a45a-8be92a62f194.png)

**This is code repeating**

![image](https://user-images.githubusercontent.com/20326858/224542358-d79210c1-669d-45ae-882f-5a17ae5700fc.png)

Reply:
I look at check list every time before I have make PR, but we can understand the one thing differently. Recomendations in the document was writen abstractly and directed to help and not give to us all answers in my opinion and made mistakes are one of part studing. 
For exemple (repeat code: a lot of part of code are repeat but as far as i understand, we should combine if parts of code are the same by logical or workflow).